src/fsharp/DotNetFrameworkDependencies.fs

```diff
@@ -470,6 +523,42 @@ module internal FSharp.Compiler.DotNetFrameworkDependencies
             yield "System.Xml.XDocument"
         ]
 
+    member _.GetDefaultReferencesForScriptsAndOutOfProjectSources (useFsiAuxLib, useDotNetFramework, useSdkRefs) =
+        fetchPathsForDefaultReferencesForScriptsAndOutOfProjectSources useFsiAuxLib useSdkRefs useDotNetFramework
+
+    member _.GetSystemAssemblies() = systemAssemblies
+
     // The set of references entered into the TcConfigBuilder for scripts prior to computing the load closure. 
-    let basicReferencesForScriptLoadClosure useFsiAuxLib useSdkRefs assumeDotNetFramework =
-        fetchPathsForDefaultReferencesForScriptsAndOutOfProjectSources useFsiAuxLib useSdkRefs assumeDotNetFramework
+    member _.GetBasicReferencesForScriptLoadClosure useFsiAuxLib useSdkRefs useDotNetFramework =
+        fetchPathsForDefaultReferencesForScriptsAndOutOfProjectSources useFsiAuxLib useSdkRefs useDotNetFramework
+
+    member _.IsInReferenceAssemblyPackDirectory filename =
+        match tryGetNetCoreFrameworkRefsPackDirectoryRoot() with
+        | _, Some root ->
+            let path = Path.GetDirectoryName(filename)
+            path.StartsWith(root, StringComparison.OrdinalIgnoreCase)
+        | _ -> false
+
+    member _.GetTfm() = getChosenTfm()
+
+    member _.GetRid() = getChosenRid()
+
+    member _.GetFrameworkRefsPackDirectory() = tryGetFrameworkRefsPackDirectory()
+
+    // Try and get a useful default .NET Core SDK directory from which to infer the target framework assemblies.
+    // If running on .NET Core we just use defaults implied by the currenly executing tooling.
+    static member TryGetDefaultSdkDirAndRid(useDotNetFramework) =
+        if useDotNetFramework || FSharpEnvironment.isRunningOnCoreClr || not (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) then
+            // We currently always use the contents inferred from where we are runing
+            None, None
+        else
+            // Running on .NET Framework 32 bit windows (e.g. devenv.exe), need to find a .NET SDK
+            let sdks = @"C:\Program Files\dotnet\sdk"  // TODO - correct this technique assuming this is devenv.exe
```

Comment:
This is kind of tough, the ide may run on the desktop ... for sure it is at the moment.  I will try and find an API, I think there may be one somewhere ...

Reply:
> This is kind of tough, the ide may run on the desktop ... for sure it is at the moment. I will try and find an API, I think there may be one somewhere ...

All the searches say "run `dotnet --info` or `dotnet--list-sdks`".  But how to find `dotnet.exe`? There must be a way for VS to probe it's corresponding dotnet install, and we can pass that through

Reply:
Is `dotnet.exe` always on PATH? This is how the official MSBuildLocator does it: https://github.com/microsoft/MSBuildLocator/blob/master/src/MSBuildLocator/DotNetSdkLocationHelper.cs